src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemSerializer.cs

```diff
@@ -39,7 +37,7 @@ public async Task<IEnumerable<DicomDataset>> DeserializeAsync(Stream stream, str
             {
                 string json = await streamReader.ReadToEndAsync().ConfigureAwait(false);
 
-                IEnumerable<DicomDataset> datasets = JsonSerializer.Deserialize<IEnumerable<DicomDataset>>(json, serializerOptions);
+                var datasets = JsonSerializer.Deserialize<T>(json, serializerOptions);
```

Comment:
what are the implications of this?

Reply:
We are trying to use the same serializer for both Add and Cancel. For Add "T" is "IEnumerable<DicomDataset>" and for Cancel "T" is DicomDataset.

I took care of updating the consumer code-segments already.